src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsDocumentFilter.cs

```diff
@@ -12,21 +12,22 @@ public class XmlCommentsDocumentFilter : IDocumentFilter
         private const string MemberXPath = "/doc/members/member[@name='{0}']";
         private const string SummaryTag = "summary";
 
+        private readonly SwaggerGeneratorOptions _options;
         private readonly XPathNavigator _xmlNavigator;
 
-        public XmlCommentsDocumentFilter(XPathDocument xmlDoc)
+        public XmlCommentsDocumentFilter(SwaggerGeneratorOptions options, XPathDocument xmlDoc)
         {
+            _options = options;
             _xmlNavigator = xmlDoc.CreateNavigator();
         }
 
         public void Apply(OpenApiDocument swaggerDoc, DocumentFilterContext context)
         {
             // Collect (unique) controller names and types in a dictionary
             var controllerNamesAndTypes = context.ApiDescriptions
-                .Select(apiDesc => apiDesc.ActionDescriptor as ControllerActionDescriptor)
-                .Where(actionDesc => actionDesc != null)
-                .GroupBy(actionDesc => actionDesc.ControllerName)
-                .Select(group => new KeyValuePair<string, Type>(group.Key, group.First().ControllerTypeInfo.AsType()));
+                .Where(apiDesc => apiDesc.ActionDescriptor is ControllerActionDescriptor)
+                .GroupBy(apiDesc => _options.TagsSelector(apiDesc).First())
+                .Select(group => new KeyValuePair<string, Type>(group.Key, ((ControllerActionDescriptor)group.First().ActionDescriptor).ControllerTypeInfo.AsType()));
```

Comment:
I think it's better with the `as` and the null check, rather than the `is` and then a cast.

Reply:
Ok, I changed it to accomplish that by selecting an anonymous type to capture both the original apiDesc as well actionDesc as well.